teps/0048-task-results-without-results.md

```diff
@@ -0,0 +1,425 @@
+  image is missing.
+
+* The third task takes the image reference from a result of the second task and deploys the container.
+
+The pipeline authors sets a default value for the image reference of it's own image reference input param, so that
+if task2 is `skipped`, task3 can deploy the image that was passed to the pipeline by the `pipelineRun`.
+
+```yaml
+kind: Pipeline
+spec:
+  params:
+    - name: targetImage
+      description: the image to deploy (and build if missing)
+  tasks:
+    - name: lookup-image
+      taskRef:
+        name: search-image
+      params:
+        - name: imageRef
+          value: "$(params.targetImage)"
+    - name: build-image
+      when:
+        - input: "$(tasks.lookup-image.results.found)"
+          operator: in
+          values: ["False"]
+      taskRef:
+        name: build-image
+      params:
+        - name: imageRef
+          value: "$(params.targetImage)"
+      results:
+        - name: builtImage
+          default: "$(params.targetImage)"
+    - name: deploy-image
+      taskRef:
+        name: deploy-image
+      params:
+        - name: imageRef
+          value: "$(tasks.build-image.results.builtImage)"
+```
+
+
+## Design Details
+
+### finally with missing task result
+
+`Pipeline` author can specify `default` for a missing result if that `param` is optional.
+
+```yaml
+spec:
+  params:
+    - name: owner-name
+      default: "the-best-owner"
+  tasks:
+   - name: boskos-acquire
+     taskRef:
+       name: boskos-acquire
+     params:
+       - name: owner-name
+         value: $(params.owner-name)
+     results:
+      - name: leased-resource
+        value: "fake-awesome-project"
+   - name: do-stuff-with-resource
+     taskRef:
+       name: do-stuff-with-resource
+     params:
+       - name: resource-name
+         value:  $(tasks.boskos-acquire.results.leased-resource)
+ finally:
+   - name: boskos-release
+     taskRef:
+       name: boskos-release
+     params:
+       - name: leased-resource
+         value:  $(tasks.boskos-acquire.results.leased-resource)
+```
+
+
+### when expression with missing task result
+
+```yaml
+spec:
+  tasks:
+    - name: add
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          value: 0
+    - name: multiply
+      when:
+        - input: "$(tasks.add.results.sum)"
+          operator: notin
+          values: ["0"]
+      taskRef:
+        name: multiply
+```
+
+### continueAfterSkip with missing task result
+
+```yaml
+spec:
+  tasks:
+    - name: add
+      when:
+        - input: "0"
+          operator: notin
+          values: ["0"]
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          value: 0
```

Comment:
What happens if `add-task` define a result name `sum` and what happens if `add-task` doesn't define a result name `sum` ?